lib/ToolbarBadgeHub.jsm

```diff
@@ -243,10 +248,26 @@ class _ToolbarBadgeHub {
     }
     let toolbarbutton = document.getElementById(message.content.target);
     if (toolbarbutton) {
+      const badge = toolbarbutton.querySelector(".toolbarbutton-badge");
+      badge.classList.add("feature-callout");
       toolbarbutton.setAttribute("badged", true);
-      toolbarbutton
-        .querySelector(".toolbarbutton-badge")
-        .classList.add("feature-callout");
+      // If we have additional aria-label information for the notification
+      // we add this content to the hidden `toolbarbutton-text` node.
+      // We then use `aria-labelledby` to link this description to the button
+      // that received the notification badge.
+      if (message.content["aria-label"]) {
+        toolbarbutton.setAttribute(
+          "aria-labelledby",
+          "toolbarbutton-notification-description"
+        );
+        toolbarbutton
+          .querySelector(".toolbarbutton-text")
+          .setAttribute("id", "toolbarbutton-notification-description");
```

Comment:
Can there only be one badged button at a time? If there can be multiple, this will be a problem because this id is global to the document and is not unique per badged button.

Reply:
Yes, [There can be only one](https://www.youtube.com/watch?v=sqcLjcSloXs).